spring-cloud-dataflow-deployer-cloudfoundry/src/main/java/org/springframework/cloud/dataflow/module/deployer/cloudfoundry/ApplicationModuleDeployer.java

```diff
@@ -317,28 +318,16 @@ private ModuleStatus notRunningInstancesStatus(ModuleDeploymentId id, CloudAppli
 	private Map<String, String> toEnvironmentVariables(HashMap<String, String> args) {
 
 		Map<String, String> env = new HashMap<>(args.size());
-		StringBuilder sb = new StringBuilder();
-		for (Map.Entry<String, String> entry : args.entrySet()) {
-			String oneArg = "--" + entry.getKey() + "=" + quoteValue(entry.getValue());
-			sb.append(bashEscape(oneArg)).append(' ');
+		ObjectMapper objectMapper = new ObjectMapper();
+		try {
+			env.put("SPRING_APPLICATION_JSON", objectMapper.writeValueAsString(args));
```

Comment:
Great! Thank you!

M-m-m. Does it really work as it is described:
> `$ SPRING_APPLICATION_JSON='{"foo":{"bar":"spam"}}' java -jar myapp.jar`

> In this example you will end up with foo.bar=spam in the Spring Environment.

I mean this your `Map -> JSON` conversion.
How does that `args` Map looks? 
And what is the output JSON?
Any samples, test?

Sorry, if I am annoying, but I wonder if we really should have something to confirm that we don't introduce an new issue...